dizoo/gym_pybullet_drones/envs/gym_pybullet_drones_env.py

```diff
@@ -0,0 +1,271 @@
+    if type_of_action in [ActionType.RPM, ActionType.DYN, ActionType.VEL]:
+        return 4
+    elif type_of_action == ActionType.PID:
+        return 3
+    elif type_of_action == ActionType.TUN:
+        return 6
+    elif type_of_action in [ActionType.ONE_D_DYN, ActionType.ONE_D_PID, ActionType.ONE_D_RPM]:
+        return 1
+    else:
+        raise ValueError('Invalid action type.')
+
+
+def gym_pybullet_drones_action_space(drone_num=1, minimum=-1, maximum=1, dtype=np.float32) -> Callable:
+
+    def _gym_pybullet_drones_action_space(type_of_action) -> Box:
+        dim = drones_action_dim(type_of_action)
+        return Box(
+            np.repeat(minimum, dim * drone_num).astype(dtype),
+            np.repeat(maximum, dim * drone_num).astype(dtype),
+            dtype=dtype
+        )
+
+    return _gym_pybullet_drones_action_space
+
+
+def gym_pybullet_drones_reward_space(minimum=-10000, maximum=0, dtype=np.float32) -> Callable:
+    return Box(np.repeat(minimum, 1).astype(dtype), np.repeat(maximum, 1).astype(dtype), dtype=dtype)
+
+
+gym_pybullet_drones_env_info = {
+    "takeoff-aviary-v0": {
+        "observation_space": gym_pybullet_drones_observation_space(12, minimum=-1, maximum=1),
+        "action_space": gym_pybullet_drones_action_space(drone_num=1, minimum=-1, maximum=1),
+        "reward_space": gym_pybullet_drones_reward_space()
+    },
+    "flythrugate-aviary-v0": {
+        "observation_space": gym_pybullet_drones_observation_space(12, minimum=-1, maximum=1),
+        "action_space": gym_pybullet_drones_action_space(drone_num=1, minimum=-1, maximum=1),
+        "reward_space": gym_pybullet_drones_reward_space()
+    },
+}
+
+
+def action_type(action_str):
+    if action_str == "PID":
+        return ActionType.PID
+    elif action_str == "DYN":
+        return ActionType.DYN
+    elif action_str == "VEL":
+        return ActionType.VEL
+    elif action_str == "RPM":
+        return ActionType.RPM
+    elif action_str == "TUN":
+        return ActionType.TUN
+    elif action_str == "ONE_D_DYN":
+        return ActionType.ONE_D_DYN
+    elif action_str == "ONE_D_PID":
+        return ActionType.ONE_D_PID
+    elif action_str == "ONE_D_RPM":
+        return ActionType.ONE_D_RPM
+    else:
+        raise ValueError('Invalid action type.')
+
+
+@ENV_REGISTRY.register('gym_pybullet_drones')
+class GymPybulletDronesEnv(BaseEnv):
+
+    def __init__(self, cfg: dict = {}) -> None:
+
+        if "num_drones" not in cfg:
+            self.env_kwargs = {
+                'drone_model': DroneModel.CF2X,
+                'initial_xyzs': None,
+                'initial_rpys': None,
+                'physics': Physics.PYB,
+                'freq': 240,
+                'aggregate_phy_steps': 1,
+                'gui': False,
+                'record': False,
+                'obs': ObservationType.KIN,
+                'act': ActionType.RPM
+            }
+        else:
+            #TODO for multi drones envs
```

Comment:
`# TODO(zjow): develop envs that support multi drones.`

Reply:
fixed.